### PR TITLE
openssl: upgrade to 1.0.2g

### PR DESCRIPTION
**Pull request by sknsean**

### CODE REVIEW

File: recipes/openssl/openssl_1.0.2g.oe [hunk 1]
```diff
@@ -1,8 +1,5 @@
 require openssl.inc
 
-# Parallel make is simply broken for now.
-PARALLEL_MAKE = ""
```
Comment (esben) [inline recipes/openssl/openssl_1.0.2g.oe]: Do you have anything concrete knowledge about parallel build fixes?  Or just re-enabling so we can find out?  Would be nice to know if we should be particularly on the look for such issues...

**Comment by esben** [comment]
:+1:

File: recipes/openssl/openssl_1.0.2g.oe (hunk 1 above)
Comment (sknsean) [inline recipes/openssl/openssl_1.0.2g.oe]: Arch does not do anything special:
https://projects.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/openssl
Yocto however have a patch:
http://git.yoctoproject.org/cgit.cgi/poky/plain/meta/recipes-connectivity/openssl/openssl/parallel.patch
I works on my builder with -j8 :-)

Comment (esben) [inline recipes/openssl/openssl_1.0.2g.oe]: Ok.  So OpenEmbedded has a patch (taken from Gentoo) for parallel build races. In my book, that clearly wins over your "works on my build with -j8" ;)

**Comment by esben** [comment]
:+1:

File: recipes/openssl/openssl_1.0.2g.oe (hunk 1 above)
Comment (kimrhh) [inline recipes/openssl/openssl_1.0.2g.oe]: for the record, I will happily accept removing disabling PARALLEL_MAKE if it has been shown to work on the oe-lite buildbot. It simply sucks that if someone finds a (maybe?) parallel make issue using $my-special-j-setting it ends up being disabled for everyone.

that said, when there is a patch, of course we chose that route

**Pull request merged**